engine/consensus/dkg/messaging_engine.go

```diff
@@ -91,7 +96,9 @@ func (e *MessagingEngine) Submit(_ network.Channel, originID flow.Identifier, ev
 
 // ProcessLocal implements the network Engine interface
 func (e *MessagingEngine) ProcessLocal(event interface{}) error {
-	return e.Process(engine.DKGCommittee, e.me.NodeID(), event)
+	return e.unit.Do(func() error {
+		return e.process(e.me.NodeID(), event)
```

Comment:
```suggestion
		err := e.process(e.me.NodeID(), event)
		if errors.Is(err, engine.IncompatibleInputTypeError) {
			// critical error - log fatal
		}
```
handle invalid input types from internal components with a crash